Helm settings update

Henotu opened the pull request:
### Missing Parameters

Missing Parameters were added to `mqtt-deployment.yaml` and `http-deployment.yaml` and the `values.yaml` respectively. The Parameter `frost.http.enableCors` was changed to `frost.http.cors.enable`.

### Removed fixed values

Fixed values inside the `*-deployment.yaml` files have been moved to `values.yaml` to be editable.
Im not sure wheter the port related settings (e.g. `{{ .Values.frost.mqtt.ports.mqtt.servicePort }}`) have been moved correctly.

### Inconsistencies
Inconsistencies naming some settings between [documentation](https://fraunhoferiosb.github.io/FROST-Server/settings/settings.html) and implemenation have been left in and new values were adapted to be that way (08fe703c67776d3c569a5a1e6936a080d38071d9) to stay consistent inside the implementation

### README
Besides the changes done in commits b4c45255...08fe703, the parameter `frost.mqtt.waitForEnter` was added to documentation because it was missing. Also, the whole table was reformatted.




#### TODO?
It could be that `enableActuation` can be removed because it isn't listed inside the [documentation](https://fraunhoferiosb.github.io/FROST-Server/settings/settings.html). It is listed in the README.md though.

Henotu committed:
[HELM]: Added CORS related settings

Henotu committed:
[HELM]: Added MQTT related settings

Henotu committed:
[HELM]: Removed constants on several occasions

Some entries inside values.yaml had to be created against the norm
inside the documentation because other already existing values were like
that

Henotu committed:
[HELM]: Added documentation to ReadMe

Besides the changes done in commits 021262f to 08fe703, the parameter
`frost.mqtt.waitForEnter` was added to documentation because it was
missing.

hylkevds commented on helm/frost-server/templates/http-deployment.yaml:
```diff
@@ -86,9 +104,9 @@ spec:
 
             # Persistence related properties
             - name: persistence_db_jndi_datasource
-              value: ""
```
`persistence_db_jndi_datasource` should always be the empty string when using docker.
JNDI is used when configuring the database using Tomcat, but in the docker case there is no way to do this, so a non-empty value makes no sense.

hylkevds commented on helm/frost-server/templates/mqtt-deployment.yaml:
```diff
@@ -34,9 +34,9 @@ spec:
           ports:
             - name: mqtt
-              containerPort: 1883
+              containerPort: {{ .Values.frost.mqtt.ports.mqtt.servicePort }}
             - name: websocket
-              containerPort: 9876
```
These two I'm pretty sure should not be changed. The `Values.frost.mqtt.ports.XXX.servicePort` setting is the external port for the service, while these variables configure the internal port outside the POD.

Essentially, for each service there are three points where a port is configured:
 - Inside the POD in the FROST code.
 - ouside the POD (the nodePort or containerPort: `Values.frost.XXX.ports.YYY.nodePort`)
 - ouside the service (the serivcePort that the client connects to: `Values.frost.XXX.ports.YYY.servicePort`)

I'm not entirely sure I got the names the right way around :)

hylkevds commented on helm/frost-server/templates/mqtt-deployment.yaml:
```diff
@@ -62,17 +62,17 @@ spec:
             - name: mqtt_mqttServerImplementationClass
               value: "de.fraunhofer.iosb.ilt.sensorthingsserver.mqtt.moquette.MoquetteMqttServer"
             - name: mqtt_Enabled
-              value: "true"
```
Setting `mqtt_Enabled` to `false` on the MQTT component makes no sense :)
Hence making this configurable is superfluous...

Henotu replied:
I'm not shure what should be done to fix this. As far as I'm concerned there are two options:
* Just set this line to `value: "true"` again. This makes both the value `.Values.frost.mqtt.enabled` and the if-clause `{{ if .Values.frost.mqtt.enabled -}}` in `http-deployment.yaml` redundant.
* Set this line to `value: true` and delete the corresponding value and if-clause from http-deployment.

hylkevds replied:
Yes and no. The environment variable on the MQTT container is redundant, since if the MQTT container is started, MQTT is obviously supposed to be turned on. This enviroment variable is only used in the all-in-one container, where the MQTT part can be turned off.

The setting `.Values.frost.mqtt.enabled` should still be there, to control if the MQTT container should be started or not. This happens in the first line in `mqtt-deployment.yaml` and `mqtt-service.yaml`:
```{{- if .Values.frost.mqtt.enabled -}}```

In the http-deployment the if on line 62 is also required, since if MQTT is enabled, a message bus is required to allow communication between the HTTP and MQTT pods, while if MQTT is not enabled, this message bus is not needed.

Henotu replied:
I haven't seen the `{{- if .Values.frost.mqtt.enabled -}}` at the start of `mqtt-deployment.yaml` - hence my confusion.
I have changed it back to "true".

hylkevds commented on helm/frost-server/templates/mqtt-deployment.yaml:
```diff
@@ -62,17 +62,17 @@ spec:
-              value: "1883"
+              value: "{{ .Values.frost.mqtt.ports.mqtt.servicePort }}"
             - name: mqtt_Host
-              value: "0.0.0.0"
```
This is incorrect. mqtt_Host is the interfaces the Java program listens on _inside the pod_. This should always be 0.0.0.0 (all interfaces). `Values.frost.mqtt.host` is the hostname outside the cluster.

hylkevds commented on helm/frost-server/templates/mqtt-deployment.yaml:
```diff
@@ -62,17 +62,17 @@ spec:
-              value: "localhost"
+              value: "{{ .Values.frost.mqtt.internalHost }}"
             - name: mqtt_WebsocketPort
-              value: "9876"
```
Using `Values.frost.mqtt.ports.websocket.servicePort` is not correct, best leave the ports alone for now.

hylkevds commented on helm/frost-server/values.yaml:
```diff
@@ -95,11 +111,14 @@ frost:
     database: sensorthings
     username: c2Vuc29ydGhpbmdz
     password: bm93eW91Y2FuY2hhbmdlaXQ=
+    jndi:
+      datasource:
```
Better to leave JNDI alone, it's a Tomcat internal thing, and makes not much sense in a Docker environment.

hylkevds reviewed (commented):
Thanks for the work! I've added some comments to specific parts.

hylkevds commented:
`enableActuation` was moved to `plugins.actuation.enable`, documented on the [plugin settings page](https://fraunhoferiosb.github.io/FROST-Server/settings/plugins.html)
For now you can leave it, if FROST finds the old setting it'll translate it to the new setting.

Henotu committed:
[HELM]: Incorporated nearly all proposed changes

* Removed configuration of JNDI
* Removed previous additions of `.Values.frost.mqtt.ports.*.servicePort`
* Configuration of `mqtt_Host` set to static again

Henotu committed:
[HELM]: Changed value of mqtt_Enabled

The value was set back to the previous "true"

hylkevds commented on helm/frost-server/templates/mqtt-deployment.yaml:
```diff
@@ -64,15 +64,15 @@ spec:
             - name: mqtt_Enabled
               value: "true"
             - name: mqtt_Port
-              value: "1883"
```
This last port shouldn't be changed to `.Values.frost.mqtt.ports.mqtt.servicePort`, same reason as the others.

Henotu replied:
My bad, it is fixed now

Henotu committed:
[HELM]: Changed value of mqtt_Port

Due to an oversight, the value was still set to
`.Values.frost.mqtt.ports.mqtt.servicePort`

hylkevds commented:
Super, thanks!
